src/sage/rings/function_field/drinfeld_modules/homset.py

```diff
@@ -305,3 +309,161 @@ def _element_constructor_(self, *args, **kwds):
+            sage: B = H.basis(3)
+            sage: M = B[0]
+            sage: M_poly = M.ore_polynomial()
+            sage: M_poly*phi.gen() - psi.gen()*M_poly
+            0
+
+        ALGORITHM:
+
+            We return the basis of the kernel of a matrix derived from the
+            constraint that `\iota \phi_T = \psi_T \iota`. See [Wes2022]_ for
+            details on this algorithm.
+        """
+        domain, codomain = self.domain(), self.codomain()
+        Fq = domain._Fq
+        K = domain.base_over_constants_field()
+        q = Fq.cardinality()
+        char = Fq.characteristic()
+        r = domain.rank()
+        n = K.degree(Fq)
+        # shorten name for readability
+        d = degree
+        qorder = logb(q, char)
+        K_basis = K.basis_over(Fq)
+        # This weird casting is done to allow extraction
+        # of coefficients in non-sparse mode from field elements
+        # There really should be a cleaner way to do this
+        pol_var = K_basis[0].polynomial().parent().gen()
+        pol_ring = PolynomialRing(Fq, str(pol_var))
+        dom_coeffs = domain.coefficients(sparse=False)
+        cod_coeffs = codomain.coefficients(sparse=False)
+        frob_matrices = []
+
+        # Compute matrices for the Frobenius operator for
+        # K over Fq
+        for order in range(d + r):
+            frob = Matrix(Fq, n, n)
+            for i, elem in enumerate(K_basis):
+                col = pol_ring(elem.frobenius(qorder*order).polynomial()) \
+                      .coefficients(sparse=False)
+                col += [0 for _ in range(n - len(col))]
+                for j in range(n):
+                    frob[j, i] = col[j]
+            frob_matrices.append(frob)
+
+        sys = Matrix(Fq, (d + r + 1)*n, (d + 1)*n)
+        for k in range(0, d + r + 1):
+            for i in range(max(0, k - r), min(k, d) + 1):
+                # We represent multiplication and Frobenius
+                # as operators acting on K as a vector space
+                # over Fq
+                # Require matrices act on the right, so we
+                # take a transpose of operators here
+                oper = K(dom_coeffs[k-i]
+                       .frobenius(qorder*i)).matrix().transpose() \
+                       - K(cod_coeffs[k-i]).matrix().transpose() \
+                       * frob_matrices[k - i]
+                for j in range(n):
+                    for l in range(n):
+                        sys[k*n + j, i*n + l] = oper[j, l]
+        sol = sys.right_kernel().basis()
+        # Reconstruct the Ore polynomial from the coefficients
+        basis = []
+        tau = domain.ore_polring().gen()
+        for basis_elem in sol:
+            basis.append(self(sum([sum([K_basis[j]*basis_elem[i*n + j]
+                               for j in range(n)])*(tau**i)
+                               for i in range(d + 1)])))
+        return basis
+
+    def random_element(self, degree, seed=None):
```

Comment:
The usual signature of `random_element` is
```python
    def random_element(self)
```
So, I think that `degree` should have a default here (I don't know nonetheless what it should be) and also, it's probably not the right place to give the possibility to define a seed.

Reply:
Actually, a good default for the degree could be $d = [K : \mathbb F_q]$ since there is always an isogeny of this degree (namely the Frobenius).

Reply:
In fact, after my next comment, probably we can just forget about `degree`.